hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/DeleteHoodieTableCommand.scala

```diff
@@ -56,32 +57,36 @@ case class DeleteHoodieTableCommand(deleteTable: DeleteFromTable) extends Runnab
   }
 
   private def buildHoodieConfig(sparkSession: SparkSession): Map[String, String] = {
-    val targetTable = sparkSession.sessionState.catalog
-      .getTableMetadata(tableId)
+    val targetTable = sparkSession.sessionState.catalog.getTableMetadata(tableId)
+    val tblProperties = targetTable.storage.properties ++ targetTable.properties
     val path = getTableLocation(targetTable, sparkSession)
     val conf = sparkSession.sessionState.newHadoopConf()
     val metaClient = HoodieTableMetaClient.builder()
       .setBasePath(path)
       .setConf(conf)
       .build()
     val tableConfig = metaClient.getTableConfig
-    val primaryColumns = HoodieOptionConfig.getPrimaryColumns(targetTable.storage.properties)
```

Comment:
@YannByron @xushiyan @danny0405 @leesf : Do we have context around why the case sensitivity was changed here. 
Looks like case sensitivity is broken w/ spark-sql Merge Into as of now. 
We are looking to work towards a fix. but wanted to ensure we don't unintentionally break something else if this piece of code was intentionally written for some reason.

Reply:
cc @boneanxs for taking a look if you have time~

Reply:
As far as I know, hudi currently uses `spark.sql.caseSensitive` to choose caseSensitive or not during analyze stage, and by default it's false, so it might be reasonable that we need to respect that configure as well here?